Faltou traduzir uma parte do documento recipes.md (#126)
gatsbyjs/gatsby-pt-BR: docs/docs/recipes.md

- [Fragmentos de imagem do Gatsby](/docs/gatsby-image/#image-query-fragments)
- [Exemplo de repositório com dados co-localizados](https://github.com/gatsbyjs/gatsby/tree/master/examples/gatsbygram)

## 7. Trabalhando com imagens

### Importar uma imagem para um componente com o webpack

As imagens podem ser importadas diretamente para um módulo JavaScript com o webpack. Esse processo reduz e copia automaticamente a imagem para a pasta `public` do seu site, fornecendo uma URL de imagem dinâmica para você passar para um elemento HTML `<img>` como um caminho de arquivo comum.

<EggheadEmbed
  lessonLink="https://egghead.io/lessons/gatsby-import-a-local-image-into-a-gatsby-component-with-webpack"
  lessonTitle="Importar uma imagem local para um componente Gatsby com webpack"
/>

#### Pré-requisitos

- Um [site Gatsby](/docs/quick-start) com um arquivo `.js` exportando um componente React
- Uma imagem (`.jpg`, `.png`, `.gif`, `.svg`, etc.) na pasta `src`

#### Instruções

1. Importe o caminho do seu arquivo em relação a pasta `src`:

```jsx:title=src/pages/index.js

> import React from 'react'

Reviewer: ```suggestion
import React from "react"
```
Apontei as aspas simples pra ficar de acordo com a parte já traduzida

// Informa ao webpack que este arquivo JS usa esta imagem

> import FiestaImg from '../assets/fiesta.jpg' // highlight-line

Reviewer: ```suggestion
import FiestaImg from "../assets/fiesta.jpg" // highlight-line
```

```

2. No `index.js`, adicione uma tag `<img>` com o `src` como o nome da importação que você usou do webpack (nesse caso `FiestaImg`), e adicione um atributo `alt` [descrevendo a imagem](https://webaim.org/techniques/alttext/):

```jsx:title=src/pages/index.js
import React from 'react'
import FiestaImg from '../assets/fiesta.jpg'

export default () => (
  // O resultado da importação é o URL da sua imagem
  <img src={FiestaImg} alt='A dog smiling in a party hat' /> // highlight-line
)
```

3. Execute `gatsby develop` para iniciar o servidor de desenvolvimento.
4. Veja sua imagem no navegador: `http://localhost:8000/`

#### Recursos adicionais

- [Exemplo de repositório importando uma imagem com o webpack](https://github.com/gatsbyjs/gatsby/tree/master/examples/recipe-webpack-image)

> - [Mais sobre todas as técnicas de imagem em Gatsby](/docs/images-and-files/)

Reviewer: ```suggestion
- [Mais sobre todas as técnicas de imagem no Gatsby](/docs/images-and-files/)
```

### Referencie uma imagem da pasta `static`

Como alternativa à importação de ativos com o webpack, a pasta `static` permite o acesso ao conteúdo que é automaticamente copiado para a pasta `public` quando o site é criado.

> Isto é uma **rota de fuga** para [casos de uso específicos](/docs/static-folder/#when-to-use-the-static-folder), e outros métodos como [importando com o webpack](#import-an-image-into-a-component-with-webpack) são recomendados para aproveitar as otimizações feitas pelo Gatsby.

Reviewer: ```suggestion
Isto é uma **rota de fuga** para [casos de uso específicos](/docs/static-folder/#when-to-use-the-static-folder), e outros métodos como [importar com o webpack](#importar-uma-imagem-para-um-componente-com-o-webpack) são recomendados para aproveitar as otimizações feitas pelo Gatsby.
```
Importar uma imagem para um componente com o webpack

│       └── fiesta.jpg
```

2. No `index.js`, adicione uma tag `<img>` com o `src` como o caminho relativo do arquivo da pasta `static`, e adicione um atributo `alt` [descrevendo a imagem](https://webaim.org/techniques/alttext/):

```jsx:title=src/pages/index.js
import React from 'react'

export default () => (
  <img src={`fiesta.jpg`} alt='A dog smiling in a party hat' /> // highlight-line
)
```

3. Execute `gatsby develop` para iniciar o servidor de desenvolvimento.
4. Veja sua imagem no navegador:`http://localhost:8000/`

#### Recursos adicionais

> - [Exemplo de repositório referenciando uma imagem da pasta estática](https://github.com/gatsbyjs/gatsby/tree/master/examples/recipe-static-image)

Reviewer 2: O nome desse folder nao deveria ficar no original? (O nome, nao o caminho)

- [Usando a Pasta Static](/docs/static-folder/)

> - [Mais sobre todas as técnicas de imagem em Gatsby](/docs/images-and-files/)

Reviewer: ```suggestion
- [Mais sobre todas as técnicas de imagem no Gatsby](/docs/images-and-files/)
```

### Otimizando e consultando imagens locais com gatsby-image

O plugin `gatsby-image` pode aliviar grande parte da dor associada à otimização de imagens no seu site.

> O Gatsby irá gerar recursos otimizados que podem ser consultados com o GraphQL e passados ​​para o componente de imagem do Gatsby. Isso cuida do trabalho pesado, incluindo a criação de vários tamanhos de imagem e o carregamento no momento certo.

Reviewer 2: Sugiro não usar tradução literal nesse trecho do heavy-lifting, pois a leitura nao fica tao natural. Exemplo de tradução não literal: 'takes care' poderia virar 'elimina' e ' of the heavy lifting' poderia virar ' da parte mais complexa'.

`)

return (
  <Img fluid={data.file.childImageSharp.fluid} alt='A corgi smiling happily' />
)
```

> c. várias imagens de um diretório (Exemplo: `images/dogs`) [filtradas](/docs/graphql-reference/#filter) pelos campos `extension` e `relativeDirectory`, e depois mapeado para os componentes `Img`

Reviewer 2: several nesse caso poderia ser 'múltiplas'

`)

return (
  <Img fluid={data.file.childImageSharp.fluid} alt='A corgi smiling happily' />
)
```

3. (Opcional) Adicione estilos embutidos ao `<Img />` como faria com outros componentes

```jsx
<Img
  fluid={data.file.childImageSharp.fluid}
  alt='A corgi smiling happily'
  style={{ border: '2px solid rebeccapurple', borderRadius: 5, height: 250 }} // highlight-line
/>
```

4. (Opcional) Forçar uma imagem na proporção desejada, substituindo o campo `aspectRatio` retornado pela consulta GraphQL antes de ser transmitida para o componente `<Img />`

```jsx
<Img
  fluid={{
    ...data.file.childImageSharp.fluid,
    aspectRatio: 1.6 // 1280 / 800 = 1.6
  }}
  alt='A corgi smiling happily'
/>
```

5. Execute `gatsby develop`, para gerar imagens de arquivos no sistema de arquivos (se ainda não o tiver feito) e armazená-las em cache

#### Recursos adicionais

- [Repositório de exemplo ilustrando estes exemplos](https://github.com/gatsbyjs/gatsby/tree/master/examples/recipes-gatsby-image)
- [API de imagem do Gatsby](/docs/gatsby-image/)

> - [Usado Gatsby Image](/docs/using-gatsby-image)

Reviewer: ```suggestion
- [Usando Gatsby Image](/docs/using-gatsby-image)
```

Post content...
```

2. Verifique se um identificador exclusivo (um slug neste exemplo) é passado no contexto quando `createPages` é chamado em `gatsby-node.js`, que posteriormente será passado para uma consulta GraphQL no componente Layout

```js:title=gatsby-node.js
exports.createPages = async ({ graphql, actions }) => {
  const { createPage } = actions

  // consulta para todas as markdowns

  result.data.allMdx.edges.forEach(({ node }) => {
    createPage({
      path: node.fields.slug,
      component: path.resolve(`./src/components/markdown-layout.js`),
      // highlight-start
      context: {

>         slug: node.fields.slug

Reviewer: ```suggestion
        slug: node.fields.slug,
```

>       }

Reviewer: ```suggestion
      },
```

      // highlight-end
    })
  })
}
```

> 3. Agora, importe `Img` de `gatsby-image` e `graphql` de `gatsby` para o componente do modelo, escreva uma [pageQuery](/docs/page-query/) para obter dados de imagem com base nos dados passados ​​no `slug` e passar esses dados para o componente `<Img />`:

Reviewer: ```suggestion
3. Agora, importe `Img` de `gatsby-image` e `graphql` de `gatsby` para o componente do modelo, escreva uma [pageQuery](/docs/page-query/) para obter dados de imagem com base nos dados passados ​​no `slug` e passe esses dados para o componente `<Img />`:
```

// highlight-end
```

4. Execute `gatsby develop`, que irá gerar imagens para os arquivos originados no sistema de arquivos

#### Recursos adicionais

- [Exemplo de repositório usando esta receita](https://github.com/gatsbyjs/gatsby/tree/master/examples/recipes-gatsby-image)
- [Imagens em destaque com frontmatter](/docs/working-with-images-in-markdown/#featured-images-with-frontmatter-metadata)
- [API Gatsby Image](/docs/gatsby-image/)

> - [UsandoGatsby Image](/docs/using-gatsby-image)

Reviewer 3: ```suggestion
- [Usando Gatsby Image](/docs/using-gatsby-image)
```

#### Recursos adicionais

- [Tutorial sobre como transformar Markdown em HTML](/tutorial/part-six/#transformer-plugins) usando `gatsby-transformer-remark`
- Procure plugins transformadores disponíveis na [biblioteca de plugins do Gatsby](/plugins/?=transformer)

## 9. Implantando seu site

Hora de mostrar. Quando estiver satisfeito com o seu site, você estará pronto para publicá-lo!

### Preparando para implantação

#### Pré-requisitos

- Um [site Gatsby](/docs/quick-start)
- A [ILC Gatsby](/docs/gatsby-cli) instalada

#### Instruções

1. Pare o servidor de desenvolvimento se estiver em execução (`Ctrl + C` na sua linha de comando na maioria dos casos)

2. Para o caminho padrão do site no diretório raiz (`/`), execute `gatsby build` usando a ILC do Gatsby na linha de comandos. Os arquivos construídos estarão agora na pasta `public`.

```shell
gatsby build
```

3. Para incluir um caminho de site diferente de `/` (tal como `/site-name/`), defina um prefixo de caminho adicionando o seguinte ao seu `gatsby-config.js` e substituindo `yourpathprefix` pelo prefixo do caminho desejado:

```js:title=gatsby-config.js
module.exports = {
  pathPrefix: `/yourpathprefix`
}
```

> Existem algumas razões para fazer isso -- por exemplo, hospedar um blog criado com o Gatsby em um domínio com outro site não criado no Gatsby. O site principal direcionaria para `example.com`, e o site Gatsby com um prefixo de caminho poderia morar em `example.com/blog`.

Reviewer 2: melhorar a traducao de 'live at'.

4. Com um prefixo de caminho definido em `gatsby-config.js`, execute `gatsby build` com a flag `--prefix-paths` para adicionar automaticamente o prefixo ao início de todos os URLs do site Gatsby e tags `<Link>`.

```shell
gatsby build --prefix-paths
```

5. Certifique-se de que o seu site tenha a mesma aparência ao executar o `gatsby build` como no `gatsby develop`. Executando `gatsby serve` ao criar seu site, você pode testar (e depurar, se necessário) o produto final antes de implantá-lo ao vivo.

```shell
gatsby build && gatsby serve
```

#### Recursos adicionais

- Caminho para criação e implantação de um site de exemplo no [tutorial parte um](/tutorial/part-one/#deploying-a-gatsby-site)
- Aprenda sobre [otimização de desempenho](/docs/performance/)

> - Leia sobre [outros tópicos relacionados à implantação](/docs/preparing-for-deployment/)

Reviewer 2: mudar a traducao de 'deploying'

> - Confira os [documentos de implantação](/docs/deploying-and-hosting/) para plataformas de hospedagem específicas e como implantar nelas

Reviewer 2: mudar a traducao de 'deploy'

> - Confira os [documentos de publicação](/docs/deploying-and-hosting/) para plataformas de hospedagem específicas e como publicarar nelas

Reviewer: ```suggestion
- Confira a [documentação de publicação](/docs/deploying-and-hosting/) para plataformas de hospedagem específicas e como publicar nelas
```

> ### Implantando em Netlify

Reviewer 2: mudar a traducao de 'deploying'

Use [`netlify-cli`](https://www.netlify.com/docs/cli/) para implantar seu aplicativo Gatsby sem sair da interface da linha de comandos.

#### Pré-requisitos

- Um [site Gatsby](/docs/quick-start) com um único componente `index.js`
- O pacote [netlify-cli](https://www.npmjs.com/package/netlify-cli) instalado
- A [ILC Gatsby](/docs/gatsby-cli) instalada

#### Instruções

1. Crie seu aplicativo gatsby usando `gatsby build`

> 2. Entre no Netlify usando `netlify login`

Reviewer 2: "Faça o login"

3. Execute o comando `netlify init`. Selecione a opção "Create & configure a new site".

4. Escolha um nome de site personalizado, se desejar, ou pressione enter para receber um nome aleatório.

5. Escolha seu [Team](https://www.netlify.com/docs/teams/).

> 6. Mude o caminho de implantação para `public/`

Reviewer 2: mudar traducao de deploy

> 7. Verifique se tudo está bem antes de implantar na produção usando `netlify deploy --prod`

Reviewer 2: melhorar a traducao de 'deploying to production'.
Sugestão:  "disponibilizar em/no ambiente de produção"

#### Recursos adicionais

> - [Hospedagem em Netlify](/docs/hosting-on-netlify)

Reviewer 2: trocar 'em' por 'na' ou 'no'

- [gatsby-plugin-netlify](/packages/gatsby-plugin-netlify)

> ### Implantando em ZEIT Now

Reviewer 2: mudar a traducao do termo 'deploy'

Use [Now CLI](https://zeit.co/download) para implantar seu aplicativo Gatsby sem sair da interface da linha de comandos.

#### Pré-requisitos

- Uma conta [ZEIT Now](https://zeit.co/signup)
- Um [site Gatsby](/docs/quick-start) com um único componente `index.js`
- Pacote [Now CLI](https://zeit.co/download) instalado

> - [ILC Gatsby](/docs/gatsby-cli) instalado

Reviewer 2: ficou ILC em um lugar e CLI em outro. Não seria o caso de padronizar?

#### Instruções

1. Faça login na Now CLI usando `now login`

2. Mude para o diretório do seu aplicativo Gatsby.js no Terminal se você ainda não estiver lá

> 3. Execute `now` para implementá-lo

Reviewer 2: Melhorar a traducao de 'deploy'

#### Recursos adicionais

> - [Implantando em ZEIT Now](/docs/deploying-to-zeit-now/)

Reviewer 2: Mudar esse termo. Poderia ser 'publicar' ou outro termo semelhante.
